src/components/App.tsx

```diff
@@ -75,30 +75,29 @@ function App() {
           <AvgTenureBar data={data} variable="latinx" />
         </ChartBlock>
         <SectionHeader title={`Pay at ${dashboardData.name} `} />
-        {/* XY heatmaps, probably */}
         <section className="viz viz-100">
           <div className="container-fluid">
             <div className="row">
               <HeatMap
-                data={data}
+                data={dataBucketed}
```

Comment:
I think the data transformation could work like this without modifying the SurveyRecord type (this isn't as neatly abstracted but the example below should illustrate what I mean).

```typescript
yLabels = enumToArray(BucketedCompensationRangeResponse);
xLabels = enumToArray(GenderResponse);
xyData = yLabels.map(yLabel => {
  return xLabels.map(xLabel => {
    // return the value for this x/y intersection of data
    const intersection = data.filter(d => d["gender"] === xLabel && d["compensation_range"] === yLabel);
    return intersection.length;
  });
});

//...//

<HeatMap data={xyData} xLabels={xLabels} yLabels={yLabels} />
```

Reply:
My mistake: since you're changing buckets on the fly the `d["compensation_range"] === yLabel` will have to be  `bucketCompensationTranslation[d["compensation_range"]] === yLabel`